RoFILib/configuration/Configuration.cpp

```diff
@@ -347,17 +347,23 @@ bool Configuration::removeSpanningEdge(ID parent, ID child) {
     }
 
     spanningPred[child] = {};
-    for (const auto& edgeOpt : spanningCross.at(child)) {
+    for (auto& edgeOpt : spanningCross.at(child)) {
         if (!edgeOpt.has_value())
             continue;
         ID nextId = edgeOpt.value().id2();
+        if (spanningPred[nextId].has_value()
+            && spanningPred[nextId].value().first == child)
+            continue;
         for (int i = 0; i < 6; ++i) {
             auto& succOpt = spanningSucc[nextId][i];
             if (succOpt.has_value())
                 continue;
             succOpt = {reverse(edgeOpt.value())};
             spanningSuccCount[nextId] += 1;
             spanningPred[child] = {std::make_pair(nextId, edgeOpt.value().side1())};
+            int toDeleteIndex = edgeOpt.value().side2() * 3 + edgeOpt.value().dock2();
```

Comment:
Just a small technical note - instead of direct math, I would prefer to hide in a function.